benchmark/storage/sql_table_benchmark.cpp

```diff
@@ -107,9 +124,131 @@ BENCHMARK_DEFINE_F(SqlTableBenchmark, SimpleInsert)(benchmark::State &state) {
   state.SetItemsProcessed(state.iterations() * num_inserts_);
 }
 
-// Insert the num_inserts_ of tuples into a DataTable in a single thread
+// Insert the num_inserts_ of tuples into a SqlTable concurrently
+// NOLINTNEXTLINE
+BENCHMARK_DEFINE_F(SqlTableBenchmark, ConcurrentInsert)(benchmark::State &state) {
+  // NOLINTNEXTLINE
+  for (auto _ : state) {
+    auto workload = [&](uint32_t id) {
+      // We can use dummy timestamps here since we're not invoking concurrency control
+      transaction::TransactionContext txn(transaction::timestamp_t(0), transaction::timestamp_t(0), &buffer_pool_,
+                                          LOGGING_DISABLED);
+      for (uint32_t i = 0; i < num_inserts_ / num_threads_; ++i) {
+        table_->Insert(&txn, *redo_, storage::layout_version_t(0));
+      }
+    };
+    common::WorkerPool thread_pool(num_threads_, {});
+    MultiThreadTestUtil::RunThreadsUntilFinish(&thread_pool, num_threads_, workload);
+  }
+
+  state.SetItemsProcessed(state.iterations() * num_inserts_);
+}
+
+// Read the num_reads_ of tuples in a sequential order from a SqlTable in a single thread
+// The SqlTable has only one schema version
+// NOLINTNEXTLINE
+BENCHMARK_DEFINE_F(SqlTableBenchmark, SingleVersionSequentialRead)(benchmark::State &state) {
+  // Populate read_table_ by inserting tuples
+  // We can use dummy timestamps here since we're not invoking concurrency control
+  transaction::TransactionContext txn(transaction::timestamp_t(0), transaction::timestamp_t(0), &buffer_pool_,
+                                      LOGGING_DISABLED);
+  std::vector<storage::TupleSlot> read_order;
+  for (uint32_t i = 0; i < num_reads_; ++i) {
+    read_order.emplace_back(table_->Insert(&txn, *redo_, storage::layout_version_t(0)));
```

Comment:
In this case wouldn't the insert time also be included in the benchmark? Wouldn't it be more accurate to move to some other set up function that isn't included in the benchmark time?

Reply:
The process time only includes code inside 
```
for (auto _ : state) {
    ...
}
```
Basically, the final benchmark number is calculated by 
`state.SetItemsProcessed(state.iterations() * num_inserts_);` divided by 
the time spent running the loop

Reply:
google benchmark might run each benchmark multiple times to get a more accurate result.

Reply:
Ah, thanks for clarifying that.